### DISCUSSION TITLE
How to specify the stale time of RSC payloads by page

### DISCUSSION THREAD

**Discussion by kiminno** [vercel/next.js#69933] [Help]
### Summary

We want to cache the RSC payload of a single dynamic route for 5 minutes, leaving all of the other dynamic routes to be cached for 30 seconds. Based on the [NextJS documentation on staleTime](https://nextjs.org/docs/app/api-reference/next-config-js/staleTimes), we can configure it globally. However, how can we configure it per page?

We have also tried performing a full prefetch of the route on page load to take advantage of the default `staleTime` of 5 minutes ([source](https://nextjs.org/docs/app/building-your-application/caching#client-side-router-cache)). However, although we successfully cache the RSC payload for 5 minutes with this method, it triggers an additional RSC payload fetch - once from the app router push and once from the prefetch with a different SHA.

What is the recommended way to cache the RSC payload of a single dynamic route for 5 minutes?
If this is not possible, what changes can we make to support it?

**Comment by CaptainN** [comment]
I'm not 100% sure here - but I think that's all derived from your internal fetch request timeouts. It's a little unclear how fetch revalidate settings make their way to the FE in the docs.

**Comment by kiminno** [comment]
Are there any alternatives that don't involve server caching?
